gravitee-am-ui/src/app/resolvers/alert-notifier.resolver.ts

```diff
@@ -14,13 +14,13 @@
  * limitations under the License.
  */
 import { Injectable } from '@angular/core';
-import { ActivatedRouteSnapshot, Resolve } from '@angular/router';
+import { ActivatedRouteSnapshot } from '@angular/router';
 import { Observable } from 'rxjs';
 
 import { AlertService } from '../services/alert.service';
 
 @Injectable()
-export class AlertNotifierResolver implements Resolve<any> {
```

Comment:
Resolve should be replaced with ResolveFn
https://blog.bitsrc.io/how-ive-replaced-deprecated-resolvers-in-angular-16-59811f79cd5b

Reply:
There's nothing not working with that yet, it's just deprecated. And yes, the new good way of doing things, but it will still be possible to keep the way things are done with a class. 

I haven't had the courage to rewrite all of theme. It's possible to take our time to do it 🙈 .

Reply:
Maybe you didn't know? 

I converted them to APIM because there aren't many of them and I know how they work. Here there are a lot of them and I'm not very familiar with how they work (fear of adding regression). Cockpit didn't convert them either

@mstasgravitee Do you want to convert this directly into a functional convention?

Reply:
If you are sure that it will not break something that's ok.

Reply:
I'm as sure as you are. All the tests I've done don't seem to show any errors on it. But it's always good to have a 2nd check.

Reply:
a yes and it wasn't me who removed the interface but the angular update script 🤷‍♂️